modules/swagger-codegen/src/main/java/io/swagger/codegen/CodegenModel.java

```diff
@@ -39,7 +39,7 @@
     public Set<String> allMandatory;
 
     public Set<String> imports = new TreeSet<String>();
-    public boolean hasVars, emptyVars, hasMoreModels, hasEnums, isEnum, hasRequired, hasOptional, isArrayModel, hasChildren;
+    public boolean hasVars, emptyVars, hasMoreModels, hasEnums, isEnum, hasRequired, hasOptional, isArrayModel, hasChildren, isInteger;
```

Comment:
It doesn't make sense to have `isInteger` in the `CodegenModel`. What about float? double? long? boolean?

`isInteger` is part of `CodegenProperty` and `CodegenParameter` and this is where it belong IMHO.

I didn't check the rest of the commit because I think this is a major problem.

Reply:
I agree with you that my implementation is not optimal. As you say, the problem is really that most `CodegenProperty`s can appear as a `CodegenModel` by redefining inline properties as definitions in the swagger file (cf. [Swagger Spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#schemaObject) ). I can think of two solutions that would be a bit nicer but also a lot more work:
1) Factor most fields (e.g. isInteger, isDouble, minimum, minLength,...) of `CodegenProperty` into a base class, `BASE_CLASS`, and make `CodegenModel` inherit from `BASE_CLASS`.
2) Instead of making `CodegenModel` inherit from `BASE_CLASS`, add a field to `CodegenModel` of type `BASE_CLASS` that contains the `isInteger`, `isLong`, etc. attributes. This field would be mostly of importance if `CodegenModel.isEnum==true` and could be zero otherwise.